cmdstanpy/install_cmdstan.py

```diff
@@ -150,51 +143,77 @@ def build(verbose: bool = False) -> None:
         )
 
 
-def compile_example() -> None:
+# pylint: disable=no-self-use
+def _wrap_build_progress_hook(
+    cmdstan_version: str,
+) -> Optional[Callable[[str], None]]:
+    """Sets up tqdm callback for CmdStan sampler console msgs."""
+    pad = ' ' * 20
+    pbar: Any = tqdm(
+        total=200,
+        bar_format="{desc} ({elapsed}) | {bar} | {postfix[0][value]}",
+        postfix=[dict(value=f'Building {cmdstan_version} {pad}')],
+        colour='blue',
+        desc='',
+        position=0,
+    )
+    success = f'--- CmdStan {cmdstan_version} built ---'
+
+    def build_progress_hook(line: str) -> None:
+        if line == '__DONE__' or line == success:
+            pbar.set_description('Done')
+            pbar.postfix[0]["value"] = success
+            pbar.update(100)
+            pbar.close()
+        else:
+            if line.startswith('--'):
+                pbar.postfix[0]["value"] = line
+            else:
+                pbar.postfix[0]["value"] = f'{line[:8]} ... {line[-20:]}'
+            pbar.set_description('Compiling')
+            pbar.update(1)
+
+    return build_progress_hook
+
+
+def compile_example(verbose: bool = False) -> None:
     """
     Compile the example model.
-    The current directory must be a cmdstan library.
+    The current directory must be a cmdstan installation, i.e.,
+    contains the makefile, Stanc compiler, and all libraries.
+
+    :param verbose: Boolean value; when ``True``, show output from make command.
     """
     cmd = [
         MAKE,
         Path(
             os.path.join('examples', 'bernoulli', 'bernoulli' + EXTENSION)
         ).as_posix(),
     ]
-    proc = subprocess.Popen(
-        cmd,
-        cwd=None,
-        stdin=subprocess.DEVNULL,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        env=os.environ,
-    )
-    while proc.poll() is None:
-        if proc.stdout:
-            proc.stdout.readline().decode('utf-8')
-    _, stderr = proc.communicate()
-    if proc.returncode:
-        msgs = ['Failed to compile example model bernoulli.stan']
-        if stderr:
-            msgs.append(stderr.decode('utf-8').strip())
-        raise CmdStanInstallError('\n'.join(msgs))
-
-
-def rebuild_cmdstan(verbose: bool = True) -> None:
+    try:
+        if verbose:
+            do_command(cmd)
+        else:
+            do_command(cmd, fd_out=io.StringIO())
+    except RuntimeError as e:
+        # pylint: disable=raise-missing-from
+        raise CmdStanInstallError(f'Command "make clean-all" failed\n{e}')
+
+
+def rebuild_cmdstan(cmdstan_version: str, verbose: bool = False) -> None:
```

Comment:
Sorry, missed this on first review - I think requiring the version argument to rebuild is pretty un-ergonomic (users might not know the exact version, and supplying the wrong version doesn't do anything besides break the progress bar, right?)

Reply:
Also, I think `rebuild` should take a progress argument now and pass it along. It can default to True

Reply:
thanks for this - completely agree - it was a hack and now it's better.